NetSerializer/TypeSerializers/DictionarySerializer.cs

```diff
@@ -148,6 +148,11 @@ static MethodInfo GetGenReader(Type containerType, Type genType)
 
 		public static void WritePrimitive<TKey, TValue>(Serializer serializer, Stream stream, Dictionary<TKey, TValue> value)
 		{
+			if (value == null)
+			{
+				return;
```

Comment:
You need to explicitly serialize null here or else you'll misalign during reading/writing, I think.

Reply:
Fixed.